torchvision/datasets/mnist.py

```diff
@@ -141,9 +147,26 @@ def download(self) -> None:
         os.makedirs(self.processed_folder, exist_ok=True)
 
         # download files
-        for url, md5 in self.resources:
-            filename = url.rpartition('/')[2]
-            download_and_extract_archive(url, download_root=self.raw_folder, filename=filename, md5=md5)
+        for filename, md5 in self.resources:
+            for mirror in self.mirrors:
+                url = "{}{}".format(mirror, filename)
+                try:
+                    print("Downloading {}".format(url))
+                    download_and_extract_archive(
+                        url, download_root=self.raw_folder,
+                        filename=filename,
+                        md5=md5
+                    )
+                except URLError as error:
+                    print(
+                        "Failed to download (trying next):\n{}".format(error)
+                    )
+                    continue
+                finally:
+                    print()
```

Comment:
These seem like debug statements, no?

```suggestion
```

Reply:
I think this might be here to avoid the download bar stopping at the middle of the screen for some reason. I'd be happy to remove it in a follow-up commit if this is unnecessary, but I'll be moving forward with merging this as this is blocking a few other things